internal/flow/flow.go

```diff
@@ -46,6 +47,18 @@ type Service struct {
 	NotifyReleaseHook func(ctx context.Context, options NotifyReleaseOptions)
 }
 
+type Storage interface {
+	ArtifactExists(ctx context.Context, artifactID string) (bool, error)
+
+	ArtifactSpecification(ctx context.Context, service, artifactID string) (artifact.Spec, error)
+	ArtifactPaths(ctx context.Context, service, environment, branch, artifactID string) (specPath, resourcesPath string, close func(context.Context), err error)
```

Comment:
Docs would help here, since I don't find the naming too obvious.
What does `Paths` methods do?

Reply:
I've added documentation to the interface methods for better understanding.